sasmodels/models/micromagnetic_FF_3D.py

```diff
@@ -0,0 +1,200 @@
+              ["mag_radius", "Ang", 50., [0, inf], "volume", "Magnetic radius of the core"],
+              ["mag_thickness", "Ang", 40., [0, inf], "volume", "Magnetic thickness of shell"], 
+              ["hk_radius", "Ang", 50., [0, inf], "volume", "Anisotropy radius of the core"],            
+              ["nuc_sld_core", "1e-6/Ang^2", 1.0, [-inf, inf], "", "Core scattering length density"],
+              ["nuc_sld_shell", "1e-6/Ang^2", 1.7, [-inf, inf], "", "Scattering length density of shell"],
+              ["nuc_sld_solvent", "1e-6/Ang^2", 6.4, [-inf, inf], "", "Solvent scattering length density"],
+              ["mag_sld_core",    "1e-6/Ang^2", 1.0,  [-inf, inf], "",    "Magnetic scattering length density of core"],
+              ["mag_sld_shell", "1e-6/Ang^2", 1.7, [-inf, inf], "", "Magnetic scattering length density of shell"],
+              ["mag_sld_solvent", "1e-6/Ang^2", 3.0,  [-inf, inf], "",  "Magnetic scattering length density of solvent"],
+              ["hk_sld_core", "1e-6/Ang^2", 1.0,  [0, inf], "",    "Anisotropy field of defect"],
+              ["Hi", "T", 2.0,  [0, inf], "",    "Effective field inside the material"],
+              ["Ms", "T", 1.0,  [0, inf], "",    "Volume averaged saturation magnetisation"], 
+              ["A", "pJ/m", 10.0,  [0, inf], "",    "Average exchange stiffness constant"],  
+              ["D", "mJ/m^2", 0.0,  [0, inf], "",    "Average DMI constant"],                          
+              ["up_i",          "None",  0.5, [0, 1], "", "Polarisation incoming beam"],
+              ["up_f",          "None",  0.5, [0, 1], "", "Polarisation outgoing beam"],
+              ["alpha",          "None",  90, [0, 180], "", "Inclination of field to neutron beam"],
+              ["beta",          "None",  0, [0, 360], "", "Rotation of field around neutron beam"],             
+             ]
+# pylint: enable=bad-whitespace, line-too-long
+
+
+
+
+source = ["lib/sas_3j1x_x.c", "lib/core_shell.c", "lib/gauss76.c", "lib/magnetic_functions.c", "micromagnetic_FF_3D.c"]
+structure_factor = False
+have_Fq = False
+single=False
+
+
+def random():
+    """Return a random parameter set for the model."""
+    outer_radius = 10**np.random.uniform(1.3, 4.3)
+    # Use a distribution with a preference for thin shell or thin core
+    # Avoid core,shell radii < 1
+    radius = np.random.beta(0.5, 0.5)*(outer_radius-2) + 1
+    thickness = outer_radius - radius
+    pars = dict(
+        nuc_radius=radius,
+        nuc_thickness=thickness,
+        mag_radius=radius,
+        mag_thickness=thickness,
+        hk_radius=radius
+        
+    )
+    return pars
+
+
+
+
+# tests = [
+    # [{'radius': 20.0, 'thickness': 10.0}, 0.1, None, None, 30.0, 4.*pi/3*30**3, 1.0],
+
+    # # The SasView test result was 0.00169, with a background of 0.001
+    # [{'radius': 60.0, 'thickness': 10.0, 'sld_core': 1.0, 'sld_shell': 2.0,
+      # 'sld_solvent': 3.0, 'background': 0.0}, 0.4, 0.000698838],
+# ]
```

Comment:
Are there any known tests that could be run on this model?

Reply:
Includes test for default values now